core/src/test/scala/org/apache/spark/util/collection/FixedHashObject.scala

```diff
@@ -22,4 +22,8 @@ package org.apache.spark.util.collection
  */
 case class FixedHashObject(v: Int, h: Int) extends Serializable {
   override def hashCode(): Int = h
+  override def equals(other: Any): Boolean = other match {
```

Comment:
Actually if we do call `super.equals()` it fails tests as case classes have there implementation of equals so we need to rewrite the same equals back again.